src/api.rs

```diff
@@ -80,6 +90,40 @@ impl API {
     };
   }
 
+  fn login(){
+    // Get credentials
+    let mut username = match env::var("CLINK_USERNAME") {
+      Ok(user) => {
+        if user.chars().all(char::is_alphanumeric) {
+          user
+        } else {
+          panic!("CSH Username Invalid! Must only contain alphanumeric symbols");
```

Comment:
ummmm

Reply:
Instead of panic, should I just default to the system user name?

Reply:
You won't need this if you use the syntax I suggested below.... You really should not be calling `sh -c` because it's very dangerous